teuthology/task/clock.py

```diff
@@ -30,8 +30,9 @@ def task(ctx, config):
     """
 
     log.info('Syncing clocks and checking initial clock skew...')
-    for rem in ctx.cluster.remotes.keys():
-        rem.run(
+
+    run.wait(
+        ctx.cluster.run(
```

Comment:
Suggestion: maybe instead of run.wait combination use the #1695 and have:
```
ctx.cluster.run(
  args=[...],
  timeout = 360,
  sync=True
)
```
Or we can make the ctx.cluster.run process remotes synchronous and wait by default.